source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc

```diff
@@ -144,6 +146,10 @@ Http::FilterHeadersStatus ProxyFilter::decodeHeaders(Http::RequestHeaderMap& hea
 
     auto const& host = config_->cache().getHost(headers.Host()->value().getStringView());
     if (host.has_value()) {
+      if (!host.value()->address()) {
+        onDnsResolutionFail();
+        return Http::FilterHeadersStatus::StopIteration;
```

Comment:
Consider calling `latchTime(decoder_callbacks_, DNS_END);` before this line (or in `onDnsResolutionFail()`.